src/GoTrueClient.ts

```diff
@@ -703,7 +718,7 @@ export default class GoTrueClient {
     if (expiresAt) {
       const timeNow = Math.round(Date.now() / 1000)
       const expiresIn = expiresAt - timeNow
-      const refreshDurationBeforeExpires = expiresIn > 60 ? 60 : 0.5
+      const refreshDurationBeforeExpires = expiresIn > EXPIRY_MARGIN ? EXPIRY_MARGIN : 0.5
       this._startAutoRefreshToken((expiresIn - refreshDurationBeforeExpires) * 1000)
     }
 
```

Comment:
I have an open issue that the first part addresses the .5 seconds as not enough time to complete refresh.
https://github.com/supabase/gotrue-js/issues/272

I have a trace showing failure to do preflight and refresh request within 500msec in that issue (even though who ever came up with it felt it was enough time)

So if it is possible for the time left to expiration is EXPIRY_MARGIN + .5 or 1 or less, the token will expire for a brief amount of time if code can run and use the token before this task is finished (which I have not traced thru entirely).

I also think the rounding error mentioned in that still exists here and in by cutting it so close with the .5 as Date.now() could get rounded up adding up to .5 seconds you don't have if on the margin.

Reply:
We've made the following decisions:

1. We will default JWT expiry to 15mins in the future (once we've verified all the token refresh issues have been resolved)
2. We will allow a minimum JWT expiry of 30s.
3. We will default the expiry margin to 10s.
4. In case of network failure we will retry every second. 

So if you set JWT to the smallest possible value (30s), the client library will refresh it every 20s.